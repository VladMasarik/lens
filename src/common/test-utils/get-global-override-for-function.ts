/**
 * Copyright (c) OpenLens Authors. All rights reserved.
 * Licensed under MIT License. See LICENSE in root directory for more information.
 */
import type { Injectable } from "@ogre-tools/injectable";
import { getGlobalOverride } from "./get-global-override";

export const getGlobalOverrideForFunction = (
  injectable: Injectable<Function, any, any>,
) =>
  getGlobalOverride(injectable, () => () => {
    throw new Error(`Tried to invoke a function "${injectable.id}" without override`);
  });
