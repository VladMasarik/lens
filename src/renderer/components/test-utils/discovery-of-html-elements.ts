/**
 * Copyright (c) OpenLens Authors. All rights reserved.
 * Licensed under MIT License. See LICENSE in root directory for more information.
 */
import type { RenderResult } from "@testing-library/react";

export const querySingleElement =
  (attributeName: string, attributeValue: string) =>
    (rendered: RenderResult) => {
      const dataAttribute = `data-${attributeName}-test`;

      return rendered.baseElement.querySelector(
        `[${dataAttribute}="${attributeValue}"]`,
      );
    };

export const queryAllElements =
  (attributeName: string) => (rendered: RenderResult) => {
    const dataAttribute = `data-${attributeName}-test`;

    const results = [...rendered.baseElement.querySelectorAll(`[${dataAttribute}]`)];

    return {
      elements: results,
      attributeValues: results.map(result => result.getAttribute(dataAttribute)),
    };
  };

export const getSingleElement =
  (attributeName: string, attributeValue: string) =>
    (rendered: RenderResult) => {
      const dataAttribute = `data-${attributeName}-test`;

      const element = querySingleElement(attributeName, attributeValue)(rendered);

      if (!element) {
        const validValues = queryAllElements(attributeName)(rendered).attributeValues;

        throw new Error(
          `Couldn't find HTML element with attribute "${dataAttribute}" with value "${attributeValue}". Valid values are:\n\n"${validValues.join(
            '",\n"',
          )}"`,
        );
      }

      return element;
    };
