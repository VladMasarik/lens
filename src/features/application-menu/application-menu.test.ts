/**
 * Copyright (c) OpenLens Authors. All rights reserved.
 * Licensed under MIT License. See LICENSE in root directory for more information.
 */
import type { DiContainer } from "@ogre-tools/injectable";
import type { ApplicationBuilder } from "../../renderer/components/test-utils/get-application-builder";
import { getApplicationBuilder } from "../../renderer/components/test-utils/get-application-builder";
import populateApplicationMenuInjectable from "./main/populate-application-menu.injectable";
import { advanceFakeTime, useFakeTime } from "../../common/test-utils/use-fake-time";

describe("application-menu", () => {
  let builder: ApplicationBuilder;
  let mainDi: DiContainer;
  let populateApplicationMenuMock: jest.Mock;

  beforeEach(async () => {
    useFakeTime();

    populateApplicationMenuMock = jest.fn();

    builder = getApplicationBuilder();

    builder.beforeApplicationStart((mainDi) => {
      mainDi.override(populateApplicationMenuInjectable, () => populateApplicationMenuMock);
    });

    // await builder.render();
    await builder.startHidden();

    mainDi = builder.mainDi;
  });

  it("when insufficient time passes, does not populate menu items yet", () => {
    advanceFakeTime(99);

    expect(populateApplicationMenuMock).not.toHaveBeenCalled();
  });

  describe("given enough time passes", () => {
    beforeEach(() => {
      advanceFakeTime(100);
    });

    it("populates application menu", () => {
      expect(populateApplicationMenuMock).toHaveBeenCalledWith(expect.any(Array));
    });

    it("populates application menu lol", () => {
      expect(populateApplicationMenuMock.mock.calls).toMatchSnapshot();
    });
  });
});
