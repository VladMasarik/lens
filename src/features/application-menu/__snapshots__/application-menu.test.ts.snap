// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application-menu given enough time passes populates application menu 1`] = `
Array [
  "root",
  "root.primary-for-mac",
  "root.primary-for-mac.about",
  "root.primary-for-mac.separator-1",
  "root.primary-for-mac.navigate-to-preferences",
  "root.primary-for-mac.navigate-to-extensions",
  "root.primary-for-mac.separator-2",
  "root.primary-for-mac.services",
  "root.primary-for-mac.separator-3",
  "root.primary-for-mac.hide",
  "root.primary-for-mac.hide-others",
  "root.primary-for-mac.unhide",
  "root.primary-for-mac.separator-4",
  "root.primary-for-mac.quit",
  "root.file",
  "root.file.add-cluster",
  "root.file.separator-1-for-file",
  "root.file.close-window",
  "root.edit",
  "root.edit.undo",
  "root.edit.redo",
  "root.edit.separator-1-in-edit",
  "root.edit.cut",
  "root.edit.copy",
  "root.edit.paste",
  "root.edit.delete",
  "root.edit.separator-2-in-edit",
  "root.edit.selectAll",
  "root.view",
  "root.view.navigate-to-catalog",
  "root.view.open-command-palette",
  "root.view.separator-1-for-view",
  "root.view.go-back",
  "root.view.go-forward",
  "root.view.reload",
  "root.view.toggle-dev-tools",
  "root.view.separator-2-for-view",
  "root.view.reset-zoom",
  "root.view.zoom-in",
  "root.view.zoom-out",
  "root.view.separator-3-for-view",
  "root.view.toggle-full-screen",
  "root.help",
  "root.help.navigate-to-welcome",
  "root.help.open-documentation",
  "root.help.open-support",
]
`;
